File: backend/internal/models/achievement.go
package models

import "time"

type Achievement struct {
	ID          string    `json:"id"`
	Title       string    `json:"title"`
	Subtitle    *string   `json:"subtitle"`
	Description string    `json:"description"`
	Theme       string    `json:"theme"`
	ImageURL    *string   `json:"imageUrl"`
	Points      int       `json:"points"`
	CreatedAt   time.Time `json:"createdAt"`
}

// AchievementWithStatus — ачивка из каталога + статус разблокировки для конкретного юзера.
// UserAchievementID == nil означает что ачивка не разблокирована.
type AchievementWithStatus struct {
	Achievement
	UserAchievementID *string `json:"userAchievementId"`
	Unlocked          bool    `json:"unlocked"`
	UnlockedAt        *string `json:"unlockedAt"`
	PersonalNote      *string `json:"personalNote"`
}

type ActivateCodeRequest struct {
	Code string `json:"code"`
}

type UpdateNoteRequest struct {
	Note string `json:"note"`
}

File: src/types/achievement.ts
import type { AchievementTheme } from '@/constants/achievementThemes';

export type { AchievementTheme };

export interface Achievement {
	id: string;
	title: string;
	subtitle?: string;
	description: string;
	theme: AchievementTheme;
	imageUrl?: string;
	points: number;
	createdAt: string;
	userAchievementId?: string;
	unlocked: boolean;
	unlockedAt?: string;
	personalNote?: string;
}

export interface AchievementComment {
	id: string;
	userId: string;
	userDisplayName: string;
	userAvatar?: string;
	userAchievementId?: string;
	achievementTypeId?: string;
	text: string;
	createdAt: string;
	modifiedAt?: string;
	replyToId?: string;
}
